compiler/next/lib/uast/AggregateDecl.cpp

```diff
@@ -30,8 +30,9 @@ bool AggregateDecl::validAggregateChildren(ASTListIteratorPair<Expression> it) {
     if (elt->isComment() || elt->isErroneousExpression()) {
       // OK
     } else if (elt->isDecl()) {
-      if (elt->isVariable() || elt->isFunction() ||
-          elt->isTupleDecl() || elt->isMultiDecl() ||
+      if (elt->isVariable() || elt->isFunction() || elt->isTupleDecl() ||
+          elt->isMultiDecl() ||
+          elt->isAggregateDecl() ||
```

Comment:
I personally find the heavy nesting hard to follow (I know you're just updating existing work here and thats fine). It comes to mind that maybe a function like `validAggregateChild(Expression *)` could then be used with `std::all_of(it.begin(), it.end(), validAggregateChild)`.